lib/matplotlib/axes/_base.py

```diff
@@ -2044,7 +2046,11 @@ def axis(self, *args, emit=True, **kwargs):
                 self.autoscale_view(tight=False)
                 # self.apply_aspect()
                 if s == 'equal':
-                    self.set_aspect('equal', adjustable='datalim')
+                    try:
+                        self.set_aspect('equal', adjustable='datalim')
+                    except RuntimeError:
```

Comment:
We only want to catch the above exception, right? Technically, we should make sure not to catch other RuntimeErrors here, either by inspecting the message or by using a custom exception above.

But I'm not sure if we have to be that pedantic. Catching the wrong exception does not change control flow, it could only issue a non-instructive error message.

Reply:
OK, looking back at it now, I'm not sure what this is supposed to catch.  `set_aspect` does not call `apply_aspect`, and `apply_aspect` is only called at draw time so far as I can tell.  I couldn't figure out a test for this, so lets see if anything fails just removing it again.

Reply:
Confirmed, this doesn't fail any tests - so I'm not even sure this was reachable.  Sorry I don't remember what prompted this int he first place!